EIPS/eip-2541.md

```diff
@@ -0,0 +1,273 @@
+Breaking final exponentiation hard part into the two parts (usually labeled w0 and w1)
+|w0| = 0x68f6427062e1b0b828e2c6f9f4f149496a147eaf584608da311dc0a5ef6ff10a7f6ef02c228c497824cd7cfb1e8685ab521a3d9309c6dd03ba806d298c79fc59521646d73808c511aff40fcf00000826f9440000000008c
+w0 is negative = false
+|w1| = 0xad1972339049ce762c77d5ac34cb12efc856a0853c9db94cc61c554757551c0c832ba4061000003b3de5800000000089
+```
+
+#### Fine points and encoding of base elements
+
+##### Field elements encoding:
+
+To encode points involved in the operation one has to encode elements of the base field and the extension field. Base field is 761 bit long.
+
+Base field element (Fp) is encoded as `96` bytes by performing BigEndian encoding of the corresponding (unsigned) integer. Corresponding integer **must** be less than field modulus.
+
+For elements of the quadratic extension field (Fp3) encoding is byte concatenation of individual encoding of the coefficients totaling in `3 * 96 = 288` bytes for a total encoding. For an Fp2 element in a form `el = c0 + c1 * v + c * v^2` where `v` is formal cubic non-residue and `c0`, `c1` and `c2` are Fp elements the corresponding byte encoding will be `encode(c0) || encode(c1) || encode(c2)` where `||` means byte concatenation.
+
+If encodings to not follow this spec anywhere during parsing in the precompile the precompile *must* return an error.
+
+##### Encoding of uncompressed points:
+
+Points in either G1 (in base field) or in G2 (in extension field) are encoded as byte concatenation of encodings of the `x` and `y` affine coordinates. Total encoding length for G1 point is thus `192` bytes and for G2 point is `576` bytes.
+
+##### Point of infinity encoding:
+
+Also referred as "zero point". For curves with `B != 0` point with coordinates `(0, 0)` (formal zeroes in Fp or Fp3) is *not* on the curve, so encoding of such point `(0, 0)` is used as a convention to encode point of infinity.
+
+##### Boolean encoding for subgroup checks:
+
+For subgroup checks it's required to encode whether it's required to run a subgroup check for the G1/G2 point or not. For this we encode a boolean as a *single byte* `0x00` for `false` and `0x01` for `true`.
+
+##### Encoding of scalars for multiplication operation:
+
+Group order has 377 bits so scalars for multiplication operation is redundantly encoded as `48` bytes by performing BigEndian encoding of the corresponding (unsigned) integer. Corresponding integer is **not** required to be less than or equal than main subgroup size.
```

Comment:
@shamatar to be compatible with `32`-byte 'words' of the EVM/Solidity (and to allow easy `mload` and `mstore` operations of 32-byte values in assembly), perhaps scalars should be encoded as `64` bytes instead?
Either:
- the first `16`-bytes could be _enforced_ to be all zeros; or 
- any 64-byte integer could be permitted, and precompiles would need to coerce any `64`-byte value into an element of `Fr` (where `r` is the 377-bit main subgroup order).
(I prefer the first option)

It'd be great to get your thoughts.

I'm facing the same decision making process for an EIP for the BW6-761 curve (which also has a 377-bit scalar field).